server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -168,6 +270,26 @@ public Result destroy(long programId, long blockDefinitionId) {
         .flashing("success", "Removed the visibility condition for this screen.");
   }
 
+  /** POST endpoint for deleting show-hide configurations. */
+  @Secure(authorizers = Authorizers.Labels.CIVIFORM_ADMIN)
+  public Result destroyEligibility(long programId, long blockDefinitionId) {
+    requestChecker.throwIfProgramNotDraft(programId);
+
+    try {
+      programService.removeBlockPredicate(programId, blockDefinitionId);
```

Comment:
issue (blocking): `removeBlockPredicate` removes the visibility predicate. This should call `removeBlockEligibilityPredicate` i think